allel/io/vcf_read.py

```diff
@@ -95,72 +107,112 @@ def _chunk_iter_transform(it, transformers):
         yield chunk, chunk_length, chrom, pos
 
 
+def _do_rename(it, fields, rename_fields, headers):
+
+    # normalise keys
+    rename_fields = {_normalize_field_prefix(k, headers): v
```

Comment:
No check if rename fields overwrite samples/calldata. 

```
allel.vcf_to_zarr("ALL.chr22.phase3_shapeit2_mvncall_integrated_v5a.20130502.genotypes.vcf.gz", t.name, fields=['calldata/GT', 'variants/POS', 'variants/ALT', 'variants/INFO', 'variants/SVLEN'], overwrite=True, rename_fields={"variants/MC": "calldata"}, region="22:16000000-160080000")      
```
gives a ValueError for trying to overwrite.